src/satosa/micro_services/ldap_attribute_store.py

```diff
@@ -265,7 +263,22 @@ def _ldap_connection_factory(self, config):
         if not bind_password:
             raise LdapAttributeStoreError("bind_password is not configured")
 
-        server = ldap3.Server(config["ldap_url"])
+        client_strategy_string = config["client_strategy"]
+        client_strategy_map = {
+            "SYNC": ldap3.SYNC,
+            "ASYNC": ldap3.ASYNC,
+            "LDIF": ldap3.LDIF,
+            "RESTARTABLE": ldap3.RESTARTABLE,
+            "REUSABLE": ldap3.REUSABLE,
+            "MOCK_SYNC": ldap3.MOCK_SYNC
+        }
+        client_strategy = client_strategy_map[client_strategy_string]
```

Comment:
this could also be 

```python
client_strategy = getattr(ldap3, config.get("client_strategy", ""), None)
if client_strategy is None:
    raise LdapAttributeStoreError("client_strategy is required")
```